pandas/core/indexes/base.py

```diff
@@ -3745,9 +3745,13 @@ def _reindex_non_unique(self, target):
                 # need to retake to have the same size as the indexer
                 indexer[~check] = -1
 
-                # reset the new indexer to account for the new size
-                new_indexer = np.arange(len(self.take(indexer)))
-                new_indexer[~check] = -1
+                if len(self):
```

Comment:
instead of this I would add a
```
if not len(self):
 .....
```
right before L 3734 (and make that if to an elif)